docs: Added docs for --cwd CLI option (#8156)
meltano/meltano: docs/docs/reference/command-line-interface.md

## Global Configuration

The following CLI options are available for the top-level `meltano` command:

### Current Working Directory Configuration

> - [`--cwd`](/reference/settings#clicwd) - Path to a directory containing a [ `meltano.yml` project file](/concepts/project#meltano-yml-project-file). Meltano will run as if it were started within that directory.

Reviewer: ```suggestion
- [`--cwd`](/reference/settings#clicwd) - Path to a directory containing a [`meltano.yml` project file](/concepts/project#meltano-yml-project-file). Meltano will run as if it were started within that directory.
```

> The following options are available for all `meltano` subcommands:

Reviewer: The `--cwd` is only available as a top-level CLI option, rather than for all `meltano` subcommands, so this isn't accurate. That said, the same applies to the `--log-level` and `--log-config` CLI options which are the only other things in this section, so maybe we just need to rename this from `Global Configuration` to `Top-Level Configuration`, and update the description text to read `The following CLI options are available for the top-level `meltano` command, rather than any particular sub-command.`

> ### Directory Configurations

Reviewer: ```suggestion
### Current Working Directory Configuration
```

> - [`--cwd`](/reference/settings#clicwd) - Path to the specific directory which contains the [ `meltano.yml` project file](/concepts/project#meltano-yml-project-file).

Reviewer: ```suggestion
- [`--cwd`](/reference/settings#clicwd) - Path to a directory containing a [ `meltano.yml` project file](/concepts/project#meltano-yml-project-file). Meltano will run as if it were started within that directory.
```
